src/openapi-generator.ts

```diff
@@ -568,6 +569,35 @@ export class OpenAPIGenerator {
     return undefined;
   }
 
+  private mapDiscriminator(
+    zodObjects: ZodObject<any>[],
+    discriminator: string
+  ): DiscriminatorObject | undefined {
+    // All schemas must be registered to use a discriminator
+    if (zodObjects.some(obj => obj._def.openapi?.refId === undefined)) {
+      return undefined;
+    }
+
+    const mapping = zodObjects.reduce((map, obj) => {
```

Comment:
We're already mutating the map so we can simplify the code and use a `for` or `forEach` with a `const map: Record<string, string> = {}` at the top

Reply:
Gotcha 👍 it's always a style choice but I'll switch it over 🙂